### PR TITLE
Week 3 HW

### PR DESCRIPTION
**Pull request by Saikiruthiga**
Created erd ,database for social media website 
created database and executed queries for meal sharing app

### CODE REVIEW

File: databases/week3/Meal_query.sql [hunk 1]
```diff
@@ -0,0 +1,385 @@
+-- Active: 1719747058156@@127.0.0.1@3306@Meal
+
+CREATE DATABASE Meal;
+
+DEFAULT CHARACTER SET = 'utf8mb4';
+
+use Meal;
+
+CREATE TABLE meal (
+    id INT PRIMARY KEY NOT NULL,
```
Comment (NatsGt) [inline databases/week3/Meal_query.sql:10]: If your ID is going to be a number that will continuously increment by one, you just set it up as AUTO_INCREMENT. By doing this, you don't need to specify an id each time you insert a new record.

File: databases/week3/Meal_query.sql [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+        'Italian pasta dish made with eggs, cheese (usually Pecorino Romano), pancetta, and black pepper.',
+        'Italy',
+        '2024-07-17 19:00:00',
+        40,
+        14.99,
+        '2024-07-10'
+    );
+
+CREATE TABLE reservation (
+    id INT PRIMARY KEY NOT NULL,
```
Comment (NatsGt) [inline databases/week3/Meal_query.sql:128]: Same, set ID as AUTO_INCREMENT

File: databases/week3/Meal_query.sql [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+    meal_id INT,
+    constraint meal_id_fk_1 Foreign Key (meal_id) REFERENCES meal (id),
+    stars INT NOT NULL,
+    created_date DATE NOT NULL
+);
+
+insert into
+    review
+values (
+        1,
```
Comment (NatsGt) [inline databases/week3/Meal_query.sql:187]: Same, set ID as AUTO_INCREMENT

File: databases/week3/Meal_query.sql [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+
+-- Get the meals that have good reviews
+select meal.id, meal.title, rev.title, rev.stars, rev.meal_id
+from meal
+    join (
+        select *
+        from review
+        where
+            stars > 3
+    ) as rev on meal.id = rev.meal_id;
```
Comment (NatsGt) [inline databases/week3/Meal_query.sql:397]: Its good that you practice inner queries, but I think a simple join was enough here.

File: databases/week3/Meal_query.sql [hunk 5]
```diff
@@ -0,0 +1,385 @@
...
+
+-- Get reservations for a specific meal sorted by created_date
+select meal.title, A.number_of_guests, A.created_date
+from meal
+    join (
+        select *
+        from reservation
+        where
+            meal_id = 3
+        ORDER BY created_date ASC
```
Comment (NatsGt) [inline databases/week3/Meal_query.sql:414]: Same as above

**Review by NatsGt — CHANGES_REQUESTED** [review]

File: databases/week3/Meal_query.sql (hunk 1 above)
Comment (Saikiruthiga) [reply databases/week3/Meal_query.sql:10]: Thank you for your feedback. Agreed with your point. I have modified the column with Auto_increment

File: databases/week3/Meal_query.sql (hunk 2 above)
Comment (Saikiruthiga) [reply databases/week3/Meal_query.sql:128]: Done

File: databases/week3/Meal_query.sql (hunk 3 above)
Comment (Saikiruthiga) [reply databases/week3/Meal_query.sql:187]: Done

File: databases/week3/Meal_query.sql (hunk 4 above)
Comment (Saikiruthiga) [reply databases/week3/Meal_query.sql:397]: Yes, I was just complicate the things . I could use normal join

File: databases/week3/Meal_query.sql (hunk 5 above)
Comment (Saikiruthiga) [reply databases/week3/Meal_query.sql:414]: Done.

**Pull request merged**